com.palantir.typescript/src/com/palantir/typescript/preferences/ProjectPreferenceStore.java

```diff
@@ -202,6 +251,186 @@ private synchronized void insertValue(String name) {
         }
     }
 
+    public boolean reloadTsConfigFile() {
```

Comment:
could the following few methods get broken out into its own class so that this file remains relatively simple? (just a preference store for read/write purposes with no complicated logic).